Mention Merge Queue branch protection rules (#104)
NixOS/release-wiki: src/Branch-Off.md

|`stableBranch`|Boolean|`false`|
|`supportedSystems`|Nix expression|`[ "x86_64-linux" "aarch64-linux" ]`|

> ## Nixpkgs branch protection rules

Reviewer: Nitpick: but we should avoid confusing branch protection with rulesets.

The former is fine colloquially, but in documentation they are two distinct GitHub features.

```suggestion
## Nixpkgs branch protection ruleset
```

> The same as for the jobsets applies to Nixpkgs branch protection rules, which can be created by NixOS org owners in advance.

Reviewer: ```suggestion
The same as for the jobsets applies to Nixpkgs branch protection ruleset, which can be created by NixOS org owners in advance.
```

> Currently, the Merge Queue rule needs adjustment for new release branches in `include`:

Reviewer: My wording isn't perfect either, but I think this may get the point across more clearly?

```suggestion
Currently, rulesets requiring Merge Queues cannot use wildcard patterns to "include" branches.
Instead, each branch must be explicitly listed in `includes`.

New release branches should be added to the ruleset's `includes` list before release.
```
